src/__tests__/actions.test.js

```diff
@@ -0,0 +1,34 @@
+import MockAdapter from 'axios-mock-adapter';
+import { login } from '../actions';
+import axios from '../utils/axiosInstance';
+import types from '../actions/types';
+import authUtils from '../utils/auth';
+
+const axiosMock = new MockAdapter(axios, { delayResponse: 500 });
```

Comment:
Nice implementation @codeshifu.I'd love to know how the axios-mock-adapter works

Reply:
As the name implies, `axios-mock-adapter` is a module that mocks the [axios](https://github.com/axios/axios) (or it's instance) library. It hijacks all request made using `axios` in an app and return a mock object instead of sending out the request over the wire (internet). This makes our test run a lot faster.